### PR TITLE
Refactor/1648/delete dead code

### LINKED ISSUE #1648: Clean up obsolete `invertColorRange` and `whiteColorBuildings`
# Feature request

As a developer I want to remove obsolete code. This reduces maintenance work and makes on-boarding easier. 

Through #1533 `invertColorRange` and `whiteColorBuildings` have become obsolete as described [here](https://github.com/MaibornWolff/codecharta/pull/1607#issuecomment-748213685). Therefore the code should be cleaned up to remove all occurrences

## Acceptance criteria
- All usages of `invertColorRange` and `whiteColorBuildings` are removed

## Assumptions & Exclusions
- should be straight forward cleaning up of store and store connections

### LINKED ISSUE #2072: metricDeltaSelected.component doesn't changes its color with mapColors
# Bug

## Expected Behavior

`metricDeltaSelected.component` changes its color with `storeService.getState().appSettings.mapColors.negativeDelta`

## Actual Behavior

It is hard coded to red or green

## Additional:
Similar to #1648 `invertDeltaColors` is obsolete and can / should be removed

### PR DESCRIPTION
**Pull request by shaman-apprentice**
# Remove invertColorRange, whiteColorBuildings and invertDeltaColors 

closes #1648 
closes #2072

### CODE REVIEW

File: visualization/app/codeCharta/util/usageDataTracker.ts [hunk 1]
```diff
@@ -218,7 +217,6 @@ export function trackEventUsageData(actionType: string, state: State, payload?:
 			!isActionOfType(actionType, HeightMetricActions) &&
 			!isActionOfType(actionType, ColorMetricActions) &&
 			!isActionOfType(actionType, ColorRangeActions) &&
-			!isActionOfType(actionType, InvertColorRangeActions) &&
```
Comment (shaman-apprentice) [inline visualization/app/codeCharta/util/usageDataTracker.ts:221]: @ce-bo FYI: InvertColorRangeAction was only called in tests anymore, as the mapColors are customizable now. Inverting is just switching mapColors.positive and mapColors.negative (also see #1648)

File: visualization/app/codeCharta/assets/scenarios.json [hunk 2]
```diff
@@ -30,9 +26,7 @@
   {
     "name": "Coverage",
     "settings": {
-      "appSettings": {
```
Comment (shaman-apprentice) [inline visualization/app/codeCharta/assets/scenarios.json:33]: @BridgeAR I am not sure if this setting had some effect. I compared the default `codecharta.cc.json` and could see no difference

File: visualization/app/codeCharta/util/algorithm/treeMapLayout/__snapshots__/treeMapGenerator.spec.ts.snap [hunk 3]
```diff
@@ -1293,7 +1293,7 @@ Array [
       "mcc": 10,
       "rloc": 70,
     },
-    "color": "#DDDDDD",
+    "color": "#69AE40",
```
Comment (BridgeAR) [inline visualization/app/codeCharta/util/algorithm/treeMapLayout/__snapshots__/treeMapGenerator.spec.ts.snap:1296]: The snapshot change here indicates that there's a change in the functionality. Is this new behavior definitely the expected one?

File: visualization/app/codeCharta/assets/scenarios.json (hunk 2 above)
Comment (BridgeAR) [reply visualization/app/codeCharta/assets/scenarios.json:33]: This is applied while choosing the scenarios. I guess it does indeed change the behavior a bit but I did not yet run the code to verify it.

File: visualization/app/codeCharta/util/usageDataTracker.ts (hunk 1 above)
Comment (BridgeAR) [reply visualization/app/codeCharta/util/usageDataTracker.ts:221]: This sounds good as long as buttons etc. still keep their behavior as before (also with custom configs).

File: visualization/app/codeCharta/util/algorithm/treeMapLayout/__snapshots__/treeMapGenerator.spec.ts.snap (hunk 3 above)
Comment (BridgeAR) [inline visualization/app/codeCharta/util/algorithm/treeMapLayout/__snapshots__/treeMapGenerator.spec.ts.snap:1296]: This somewhat indicates a behavior change to me. Are we certain everything is as it was before?

**Review by BridgeAR — COMMENTED** [review]
Good work! The simplification is great! We should still take a closer look at a few edge cases before merging (as outlined in my comments) but otherwise, it's LGTM.

File: visualization/app/codeCharta/util/algorithm/treeMapLayout/__snapshots__/treeMapGenerator.spec.ts.snap (hunk 3 above)
Comment (shaman-apprentice) [reply visualization/app/codeCharta/util/algorithm/treeMapLayout/__snapshots__/treeMapGenerator.spec.ts.snap:1296]: I stumbled over it as well. In https://github.com/MaibornWolff/codecharta/pull/2073/commits/00e4df0b06bce6d4a9ec4a0b9f6b9f6f53898b43 's message I explained 

> refactor: update snapshots
>
> Snapshots included obsolete whitecolor building which weren't activatable in app anymore.
Now snapshots have the default color, which reflects the truth.

File: visualization/app/codeCharta/assets/scenarios.json (hunk 2 above)
Comment (shaman-apprentice) [reply visualization/app/codeCharta/assets/scenarios.json:33]: If it does change something, I would appreciate a little demonstration :)

**Review by ngormsen — APPROVED** [review]
LGTM

File: visualization/app/codeCharta/util/algorithm/treeMapLayout/__snapshots__/treeMapGenerator.spec.ts.snap (hunk 3 above)
Comment (shaman-apprentice) [reply visualization/app/codeCharta/util/algorithm/treeMapLayout/__snapshots__/treeMapGenerator.spec.ts.snap:1296]: For triple Checking I checked out main and only set `whiteColorBuildings: true` to `false` in [dataMocks.ts](https://github.com/MaibornWolff/codecharta/blob/main/visualization/app/codeCharta/util/dataMocks.ts#L1593) leading to this exact change in snapshot tests

Comment (shaman-apprentice) [reply visualization/app/codeCharta/util/algorithm/treeMapLayout/__snapshots__/treeMapGenerator.spec.ts.snap:1296]: Same as [this](https://github.com/MaibornWolff/codecharta/pull/2073#discussion_r661286086)

**Review by BridgeAR — APPROVED** [review]
RSLGTM

**Pull request merged**